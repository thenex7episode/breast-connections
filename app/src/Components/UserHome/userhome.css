

.d-posts-wrapper{
    display:flex;
    flex-direction: column;
    justify-content: space-between;
    position: relative;

}
.d-posts-content{
    background-color: white;
    padding:1em;
    color: black;
    font-size: 1.5em;
}
.d-userhome-header{
    display:flex;
    flex-direction: row;
    background:black;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 15em;
    padding-top:5em;
}

.d-connect{
    color:white;
    font-size: 6em;
}
.d-inspire{
    color:#F433AB;
    font-size: 6em;
    
}
.d-thrive{
    color:white;
    font-size: 6em;
}
.select-box{
    margin: 0 auto;
    
}
.category-current{
    display:inline-block;
    padding: 20px;
}
.category-dropdown{
    margin-left: 1.5em;
}


@media only screen and (max-width: 1027px) {
    .d-connect, .d-inspire, .d-thrive{
        font-size: 5em;
    }
}
@media only screen and (max-width: 885px) {
    .d-connect, .d-inspire, .d-thrive{
        font-size: 4em;
    }
}
@media only screen and (max-width: 600px) {
    .d-connect, .d-inspire, .d-thrive{
        font-size: 3em;
    }
    .d-userhome-content{
        padding-top: 3.5em;
    }
}
@media only screen and (max-width: 480px) {
    .d-connect, .d-inspire, .d-thrive{
        font-size: 2em;
    }
    .d-userhome-content{
        padding-top: 3.5em;
    }
}
