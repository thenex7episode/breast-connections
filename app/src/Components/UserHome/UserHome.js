import React, { Component } from 'react';
import axios from 'axios'
import './userhome.css'
import Post from '../Post/Post'
import Sidebar from '../Sidebar/Sidebar'


class UserHome extends Component{
    constructor(){
        super()

        this.state = {

            posts: []

        }
    }
    componentDidMount(){
        axios.get('/api/posts/').then(response => {
            console.log('all posts-----------------------------',response)
            this.setState({posts: response.data})
        })
    }

    render(){
        const allPosts = this.state.posts.map((el,i) => {
            return (
              <div className='d-posts-wrapper'>
              <div className='d-posts-content'>
            <Post loggedUser={this.state.loggedUser} key={i} post_id={el.post_id} title={el.title} body={el.body} user_id={el.user_id} date={el.date} tracker={el.tracker} deletePostFn={this.deletePost}/>
                </div>
                </div>
            )
        })
        return <div>
            <div className="d-userhome-header">
                <div className='d-userhome-content'>
                    <h1 className='d-connect'>Connect.</h1>
                    <h1 className='d-inspire'>Inspire.</h1>
                    <h1 className='d-thrive'>Thrive.</h1>
                </div> 
            </div>
            
            {allPosts}
            
        </div>
     }
}
export default UserHome;