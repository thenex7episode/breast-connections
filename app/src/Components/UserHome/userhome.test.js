import UserHome from './UserHome'

it("renders correctly", () => {
    const wrapper = shallow(
        <UserHome />
    );
    expect(wrapper).toMatchSnapshot()
});

it("was clicked", () => {
    const spy = sinon.spy()
    const wrapper = mount(
        <UserHome />
    );
    const text = wrapper.find('div')
    .first()
    .simulate('click')
    expect(spy.calledOnce).toBe(false)
});