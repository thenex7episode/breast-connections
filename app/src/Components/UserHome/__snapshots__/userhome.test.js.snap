// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="d-userhome-header"
  >
    <h1
      className="d-connect"
    >
      Connect.
    </h1>
    <h1
      className="d-inspire"
    >
      Inspire.
    </h1>
    <h1
      className="d-thrive"
    >
      Thrive.
    </h1>
  </div>
  <div
    className="category-dropdown"
  >
    <Dropdown
      mouseEnterDelay={0.15}
      mouseLeaveDelay={0.1}
      overlay={
        <Menu
          className=""
          focusable={false}
          prefixCls="ant-menu"
          theme="light"
        >
          <MenuItem>
            <a
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Motivation
            </a>
          </MenuItem>
          <MenuItem>
            <a
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Resources
            </a>
          </MenuItem>
          <MenuItem>
            <a
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Family
            </a>
          </MenuItem>
          <MenuItem>
            <a
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Nutrition
            </a>
          </MenuItem>
          <MenuItem>
            <a
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Health
            </a>
          </MenuItem>
          <MenuItem>
            <a
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Financial
            </a>
          </MenuItem>
        </Menu>
      }
      placement="bottomCenter"
      prefixCls="ant-dropdown"
    >
      <Button
        ghost={false}
        loading={false}
        prefixCls="ant-btn"
      >
        Categories
      </Button>
    </Dropdown>
    <div
      className="category-current"
    >
      <button
        className="category-button"
      >
        Current Category is 
      </button>
    </div>
    <Button
      ghost={false}
      loading={false}
      onClick={[Function]}
      prefixCls="ant-btn"
    >
      Clear Category
    </Button>
  </div>
</div>
`;
